### PR TITLE
Fix for issue #1741 (Save Torrent File As...)

### PR DESCRIPTION
**Pull request by gpatarin**
Making the save file dialog sync to ensure working properly on MacOS ……and others

**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[X ] Bug fix
[ ] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
Changed the used method in `torrent-list-controller.js`

**Which issue (if any) does this pull request address?**
#1741

**Is there anything you'd like reviewers to focus on?**
Code quality (for improving)

### CODE REVIEW

File: src/renderer/controllers/torrent-list-controller.js [hunk 1]
```diff
@@ -299,15 +299,15 @@ module.exports = class TorrentListController {
...
 
-    electron.remote.dialog.showSaveDialog(win, opts, function (savePath) {
-      console.log('Saving torrent ' + torrentKey + ' to ' + savePath)
-      if (!savePath) return // They clicked Cancel
-      const torrentPath = TorrentSummary.getTorrentPath(torrentSummary)
-      fs.readFile(torrentPath, function (err, torrentFile) {
+    const savePath = electron.remote.dialog.showSaveDialogSync(win, opts)
+
+    console.log('Saving torrent ' + torrentKey + ' to ' + savePath)
+    if (!savePath) return // They clicked Cancel
```
Comment (hicom150) [inline src/renderer/controllers/torrent-list-controller.js]: I would do this check before the `console.log` 😉

File: src/renderer/controllers/torrent-list-controller.js [hunk 2]
```diff
@@ -299,15 +299,15 @@ module.exports = class TorrentListController {
       ]
     }
 
-    electron.remote.dialog.showSaveDialog(win, opts, function (savePath) {
-      console.log('Saving torrent ' + torrentKey + ' to ' + savePath)
-      if (!savePath) return // They clicked Cancel
-      const torrentPath = TorrentSummary.getTorrentPath(torrentSummary)
-      fs.readFile(torrentPath, function (err, torrentFile) {
+    const savePath = electron.remote.dialog.showSaveDialogSync(win, opts)
```
Comment (hicom150) [inline src/renderer/controllers/torrent-list-controller.js]: I would be nice if we can add `buttonLabel: 'Save'` property in `opts`

File: src/renderer/controllers/torrent-list-controller.js (hunk 1 above)
Comment (gpatarin) [reply src/renderer/controllers/torrent-list-controller.js]: Done ;)

File: src/renderer/controllers/torrent-list-controller.js (hunk 2 above)
Comment (gpatarin) [reply src/renderer/controllers/torrent-list-controller.js]: Done 😉

**Comment by hicom150** [comment]
Thank you for your contribution @gpatarin 💪 Nice work! 🎉

**Pull request merged**